src/runtime/virtcontainers/kata_agent.go

```diff
@@ -364,6 +364,11 @@ func (k *kataAgent) init(ctx context.Context, sandbox *Sandbox, config KataAgent
 	k.kmodules = config.KernelModules
 	k.dialTimout = config.DialTimeout
 
+	configuredRequestTimeout := time.Duration(sandbox.config.ImageRequestTimeout) * time.Second
+	if defaultRequestTimeout != configuredRequestTimeout {
+		defaultRequestTimeout = configuredRequestTimeout
```

Comment:
1. The `if` is not needed with the current code, since you just copy the value.
2. Consider renaming `defaultRequestTimeout`, since it's no longer a default value.
3. Should we log something when this happens? Say someone entered a value of 1 second thinking it was 1 minute? Make it evident in the log that the timeout is short for a reason.

Reply:
> 1. The `if` is not needed with the current code, since you just copy the value.

Thanks @c3d.  I'll remove the condition.

Reply:
> 2. Consider renaming `defaultRequestTimeout`, since it's no longer a default value.

Maybe we can rename it to `RequestTimeout` or `ImageRequestTimeout`?

Reply:
> 3. Should we log something when this happens? Say someone entered a value of 1 second thinking it was 1 minute? Make it evident in the log that the timeout is short for a reason.

 I agree to describe why the default timeout is short. But I haven't tested what the most appropriate timeout should be. Users should have the right to set a smaller value, if they want to get the error of the request as soon as possible.

Reply:
> > 3\. Consider renaming `defaultRequestTimeout`, since it's no longer a default value.
> 
> Maybe we can rename it to `RequestTimeout` or `grpcRequestTimeout`?

Usually, it's good if we have the same name as in the config. Makes searching the code easier. So I vote for either `ImageRequestTimeout`, or to change the config name as well. That being said, I don't see this as a 'grpc' timeout, since I think it covers just image requests, not a timeout on every single gprc request. Or does it?

Reply:
> > > 3. Consider renaming `defaultRequestTimeout`, since it's no longer a default value.
> > 
> > 
> > Maybe we can rename it to `RequestTimeout` or `grpcRequestTimeout`?
> 
> Usually, it's good if we have the same name as in the config. Makes searching the code easier. So I vote for either `ImageRequestTimeout`, or to change the config name as well. That being said, I don't see this as a 'grpc' timeout, since I think it covers just image requests, not a timeout on every single gprc request. Or does it?

`defaultRequestTimout` covers all grpc requests. So I add `imageRequestTimout` to cover image requests.